src/representative_subset_calculator/buffers/buffer_builder.h

```diff
@@ -138,7 +138,7 @@ class GlobalBufferLoader : public BufferLoader {
 
             auto index = binaryInput.begin() + rankStart;
             auto elementStop = binaryInput.begin() + rankStart;
-            while (index != rankStop && elementStop != rankStop) {
+            while (index < rankStop && elementStop < rankStop) {
```

Comment:
@ReetBarik I reset your changes to this file. The only reason why this didn't work out of the box was that `rankStart` was initialized as `displacements[rank] + 1` and when rank `0` sends zero bytes this is greater than `rankStop`. So rank 0 tries to iterate through the entire byte array.